components/drivers/misc/adc.c

```diff
@@ -64,6 +63,10 @@ static rt_err_t _adc_control(rt_device_t dev, int cmd, void *args)
             result = RT_EOK;
         }
     }
+    else if (cmd == RT_ADC_CMD_GET_VREF && adc->ops->get_resolution)
+    {
+        result = adc->ops->get_vref(adc);
```

Comment:
we cannot directly return the voltage value, the result is rt_err_t type.
```suggestion
    else if (cmd == RT_ADC_CMD_GET_VREF && adc->ops->get_vref && args)
    {
        rt_int16_t value;

        value= adc->ops->get_vref(adc);
        *((rt_int16_t*)args) = value;
        if (value == 0)
        {
            result = -RT_ENOSYS;
        }
        else
        {
            result = RT_EOK;
        }
```

Reply:
Please help me to add arg check in adc->ops->get_resolution branch

Reply:
I think rt_int16_t , not rt_uint16_t or rt_uint32_t

Reply:
all of the value of reference voltage changes from  rt_uint32_t to rt_int16_t